### PR TITLE
Network Indexer Dev Tools Page

### PR DESCRIPTION
**Pull request by walkerlj0**
…pdate to instructions

### CODE REVIEW

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+
+## <Dev Tool> Introduction
+The [IPNI network indexer](https://github.com/ipni) is a project created and maintained by Protocol Labs that is designed to index the data on the Filecoin and IPFS networks, and work alongside the Distributed Hash Table (DHT) employed by both the IPFS and Filecoin networks to enable the fast and efficient retrieval of content-addressed data.
+
+## Why Use Indexed Content Routing?
+Protocol Labs' Network Indexer (IPNI) enables any user or developer to query both the IPFS and Filecoin public nodes to find content-addresses data using the [CID](../ipfs/content-addressing/#content-addressing).
+
+Since IPFS and Filecoin use different protocols to retrieve data (IPFS uses Bitswap, and Filecion uses Graphsync), there is a need for a solution that makes it possible for anyone who would want to retrieve and use that data to locate and understand which protocol they can use to retrieve that data and use it in other applications.
+
+With versions of kubo 16 and above, a feature was implemented that makes it possible to use HTTP to send and receive information about IP addresses of peers that have certain content (CIDs) and publish IPNS records.
```
Comment (ischasny) [inline content/en/curriculum/dev-tools/network-indexer/index.md:23]: Kubo 16
```suggestion
Kubo 16 introduced Delegated Routing protocol that allows a node to publish its CIDs to alternative routing systems alongside the DHT and be used for subsequent content-addressed data lookups. 
```

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+
+The Indexer also makes it possible, in a way not achieved before, for people to retrieve data from the Filecoin network, as there is a lot of data stored across Filecoin nodes, but  no way to search and retrieve that data searching either by the CID of multihash of that data.
+
+Though the DHT is an amazing, distributed way to advertise and discover content, the IPFS indexer adds another layer that can give you other options for retrieving files for your applicaitons. With an open protocol that anyone can use, as seen here in the [go-delgated-routing](https://github.com/ipfs/go-delegated-routing) library on their server.
+to enable faster discovery and routing, along with load balancing and an alternative market where routing is done, besides the DHT.
```
Comment (ischasny) [inline content/en/curriculum/dev-tools/network-indexer/index.md:28]:
```suggestion
With an open protocol that anyone can use, as seen here in the [go-delgated-routing](https://github.com/ipfs/go-delegated-routing) library on their server.
to enable faster discovery and routing, along with load balancing and an alternative market where routing is done, besides the DHT
```
IPNI is an open protocol that anyone can use to quickly find content-addressed data on IPFS or Filecoin, to index the data that they host or to run an indexer of their own. IPNI enables faster discovery and routing, along with load balancing and an alternative market where routing is done, besides the DHT. IPNI integrates with IPFS through the [go-delegated-routing](https://github.com/ipfs/go-delegated-routing) and [index-provider](https://github.com/ipni/index-provider) libraries.

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+
+Laern more about IPFS Indexer, the motivations behind it, and potential use cases, see the video **[State of Content Routing| IPFS Camp Lisbon 2022](https://www.youtube.com/watch?v=DLCTEXbF1Es)**
+
+
+## Indexer Implementations
+Since Filecoin and IPFS store data on separate networks, using different methods of data transfer, there are who different ways that the information about the CIDs are communicated to IPNI.
```
Comment (ischasny) [inline content/en/curriculum/dev-tools/network-indexer/index.md:34]:
```suggestion
IPNI builds its index by processing (Advertisements)[https://github.com/ipni/storetheindex/blob/main/api/v0/ingest/schema/schema.ipldsch#L40]. The Advertisement construct allows a Storage Provider to offer their CIDs to IPNI, which would make these CIDs available for fast lookups once the Advertisement has been processed. Since Filecoin and IPFS store data on separate networks, using different methods of data transfer, there are who different ways that the information about the CIDs are communicated to IPNI.
```

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+
+### Filecoin Indexers
```
Comment (ischasny) [inline content/en/curriculum/dev-tools/network-indexer/index.md:36]: Title `Filecoin Indexers` might give an impression that IPFS and Filecoin are indexer by physically different systems while in reality it's not true. Maybe rename it to "Advertising from Filecoin", "Advertising from IPFS", or something along these lines?

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+Storage providers such as [Estuary](https://estuary.tech/) who store on Filecion can publish their content to the indexer using [index-provider](https://github.com/ipni/index-provider) library. When a deal is made and data is published to Filecoin, the would also [advertise](https://github.com/ipni/storetheindex/blob/main/api/v0/ingest/schema/schema.ipldsch#L40) to IPNI. This means that the CIDs and the information about the data in that CID is published on libp2p to a topic that IPNI follows.
```
Comment (ischasny) [inline content/en/curriculum/dev-tools/network-indexer/index.md:37]:
```suggestion
An update is made available on a libp2p to a topic that IPNI follows and then reaches out to the Storage Providers for an update when sees it. 
```

Comment (ischasny) [inline content/en/curriculum/dev-tools/network-indexer/index.md:37]:
```suggestion
Storage providers such as [Estuary](https://estuary.tech/) who store on Filecion can publish their content to the indexer using [index-provider](https://github.com/ipni/index-provider) library. When a deal is made and data is published to Filecoin, the would also [advertise](https://github.com/ipni/storetheindex/blob/main/api/v0/ingest/schema/schema.ipldsch#L40) to IPNI. An update is made available on a libp2p to a topic that IPNI follows and then reaches out to the Storage Providers for an update when sees it. 
```

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+
+### IPFS Indexers
+When data is added to the IPFS network via kubo version 16 and above, there is an API used with the Reframe or [go-delegated-routing](https://github.com/ipfs/go-delegated-routing) that is used to send data about those CIDs in IPNI. Anyone who published to ipfs using Kubo can configure their node to advertise CIDs into both the DHT and IPNI.
```
Comment (ischasny) [inline content/en/curriculum/dev-tools/network-indexer/index.md:40]:
```suggestion
When data is added to the IPFS network via kubo version 16 and above, there is an API used with the Reframe or [go-delegated-routing](https://github.com/ipfs/go-delegated-routing) that is used to send data about those CIDs to IPNI. Anyone who published to ipfs using Kubo can configure their node to advertise CIDs into both the DHT and IPNI.
```

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 7]
```diff
@@ -0,0 +1,323 @@
...
+
+#### Reframe
+[Reframe](https://github.com/ipfs/go-delegated-routing) is a Request-response Protocol (RPC) used by IPFS kubo 16+ that kicks off a procedure or subroutine to execute on the IPFS network that stores the data in an index, and allows others to retrieve information related to that CID.
```
Comment (ischasny) [inline content/en/curriculum/dev-tools/network-indexer/index.md:43]:
```suggestion
[Reframe](https://github.com/ipfs/go-delegated-routing) is a Request-response Protocol (RPC) used by IPFS kubo 16+ that allows the node to publish its own CIDs as well as to look up other CIDs in an alternative routing system such as IPNI alongside the DHT. That doesn't have to be either / or choice - content can be looked up in both and whichever is faster wins. 
```

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 8]
```diff
@@ -0,0 +1,323 @@
...
+With versions of kubo 16+, the Refame RPC is integrated as a feature
+* Reframe adds an additional way to discover peers, content, and IPNS records
+* With Reframe, you can configure your IPFS kubo node to publish a snapshot of all of the CIDs on your node at whatever frequency you would like
+* Reframe uses HTTP transport to store this information on an indexer node
+* Reframe is an alternative to the DHT
```
Comment (ischasny) [inline content/en/curriculum/dev-tools/network-indexer/index.md:48]: The idea behind reframe was to democratize content routing and enable a market where alternative content routing systems can emerge. We shouldn't be advertising it as a replacement for DHT - rather something that work alongside DHT and be better for some usecases.

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 9]
```diff
@@ -0,0 +1,323 @@
...
+
+### IPFS Indexers
+When data is added to the IPFS network via kubo version 16 and above, there is an API used with the Reframe or [go-delegated-routing](https://github.com/ipfs/go-delegated-routing) that is used to send data about those CIDs in IPNI. Anyone who published to ipfs using Kubo can configure their node to advertise CIDs into both the DHT and IPNI.
+
+#### Reframe
+[Reframe](https://github.com/ipfs/go-delegated-routing) is a Request-response Protocol (RPC) used by IPFS kubo 16+ that kicks off a procedure or subroutine to execute on the IPFS network that stores the data in an index, and allows others to retrieve information related to that CID.
+With versions of kubo 16+, the Refame RPC is integrated as a feature
+* Reframe adds an additional way to discover peers, content, and IPNS records
+* With Reframe, you can configure your IPFS kubo node to publish a snapshot of all of the CIDs on your node at whatever frequency you would like
+* Reframe uses HTTP transport to store this information on an indexer node
```
Comment (ischasny) [inline content/en/curriculum/dev-tools/network-indexer/index.md:47]: Reframe doesn't speak to indexers directly. Instead it relies on someone translating its messages into the IPNI protocol. Might be better to rephrase it to "Reframe is built on top of HTTP."

**Comment by ischasny** [comment]
Another thing that tutorial can talk about is [default](https://github.com/ipfs/kubo/releases/tag/v0.18.0#default-interplanetary-network-indexer) IPNI support as of Kubo b0.18.0.

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 10]
```diff
@@ -0,0 +1,324 @@
...
+With an open protocol that anyone can use, as seen here in the [go-delgated-routing](https://github.com/ipfs/go-delegated-routing) library on their server.
+to enable faster discovery and routing, along with load balancing and an alternative market where routing is done, besides the DHT
+
+Laern more about IPFS Indexer, the motivations behind it, and potential use cases, see the video **[State of Content Routing| IPFS Camp Lisbon 2022](https://www.youtube.com/watch?v=DLCTEXbF1Es)**
+
+
+## Indexer Implementations
+IPNI builds its index by processing (Advertisements)[https://github.com/ipni/storetheindex/blob/main/api/v0/ingest/schema/schema.ipldsch#L40]. The Advertisement construct allows a Storage Provider to offer their CIDs to IPNI, which would make these CIDs available for fast lookups once the Advertisement has been processed. Since Filecoin and IPFS store data on separate networks, using different methods of data transfer, there are who different ways that the information about the CIDs are communicated to IPNI.
+
+### Filecoin Indexers
```
Comment (walkerlj0) [inline content/en/curriculum/dev-tools/network-indexer/index.md:37]:
```suggestion
### Advertising Filecoin CIDs
```

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 11]
```diff
@@ -0,0 +1,324 @@
...
+Storage providers such as [Estuary](https://estuary.tech/) who store on Filecion can publish their content to the indexer using [index-provider](https://github.com/ipni/index-provider) library. When a deal is made and data is published to Filecoin, the would also [advertise](https://github.com/ipni/storetheindex/blob/main/api/v0/ingest/schema/schema.ipldsch#L40) to IPNI. This means that the CIDs and the information about the data in that CID is published on libp2p to a topic that IPNI follows.
+
+### IPFS Indexers
```
Comment (walkerlj0) [inline content/en/curriculum/dev-tools/network-indexer/index.md:40]:
```suggestion
### Advertising IPFS CIDs
```

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 12]
```diff
@@ -0,0 +1,328 @@
...
+
+### Advertising IPFS CIDs
+When data is added to the IPFS network via kubo version 16 and above, there is an API used with the Reframe or [go-delegated-routing](https://github.com/ipfs/go-delegated-routing) that is used to send data about those CIDs to IPNI. Anyone who published to ipfs using Kubo can configure their node to advertise CIDs into both the DHT and IPNI.
+
+#### Reframe
+[Reframe](https://github.com/ipfs/go-delegated-routing) is a Request-response Protocol (RPC) used by IPFS Kubo 16+ that allows the node to publish its own CIDs as well as to look up other CIDs in an alternative routing system such as IPNI alongside the DHT. That doesn't have to be either / or choice - content can be looked up or published to both in the same time.
```
Comment (walkerlj0) [inline content/en/curriculum/dev-tools/network-indexer/index.md:44]:
```suggestion
[Reframe](https://github.com/ipfs/go-delegated-routing) is a Request-response Protocol (RPC) used by IPFS Kubo 16+ that enables the translation of CIDs using the HTTP protocol to IPNI.
```

File: content/en/curriculum/dev-tools/network-indexer/index.md (hunk 9 above)
Comment (walkerlj0) [reply content/en/curriculum/dev-tools/network-indexer/index.md:52]: Updated the paragraph

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 13]
```diff
@@ -0,0 +1,328 @@
...
+When data is added to the IPFS network via kubo version 16 and above, there is an API used with the Reframe or [go-delegated-routing](https://github.com/ipfs/go-delegated-routing) that is used to send data about those CIDs to IPNI. Anyone who published to ipfs using Kubo can configure their node to advertise CIDs into both the DHT and IPNI.
+
+#### Reframe
+[Reframe](https://github.com/ipfs/go-delegated-routing) is a Request-response Protocol (RPC) used by IPFS Kubo 16+ that enables the translation of CIDs using the HTTP protocol to IPNI.
+
+Kubo doesn't know how to speak to IPNI by itself. Instead it relies on someone translating reframe messages into the IPNI protocol. That is done to enable alternative routing systems market without being bound to the DHT or IPNI protocols. 
+
+That job is done by Index Provider process that can be run alongside a Kubo node as a bridge to IPNI. 
+With versions of kubo 16+, the Refame RPC is integrated as a feature
+* Reframe adds an additional way to discover peers, content, and IPNS records
```
Comment (walkerlj0) [inline content/en/curriculum/dev-tools/network-indexer/index.md:50]:
```suggestion
* Reframe adds an additional way to discover peers, content, and IPNS records alongside the DHT
```

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 14]
```diff
@@ -0,0 +1,328 @@
...
+  * [Distributions](https://dist.ipfs.tech/#kubo)
+* [IPFS companion](https://chrome.google.com/webstore/detail/ipfs-companion/nibjojkomfdiaoajekhjakgkdhaomnch?hl=en) browser extension or [Brave browser](https://brave.com/download/) to view websites published on IPFS
+- Install a [JSON formatter extensio](https://chrome.google.com/webstore/detail/json-formatter/bcjindcccaagfpapjjmafapmmgkkhgoa?hl=en)n for your web browser
+
+## Tutorial Instructions
+You can see the video of the workshop called **[Indexed Content Routing](https://www.youtube.com/watch?v=aN7fGturjzA&t=121s)** from IPFS Camp Lisbon 2022. The original instructions are at [https://github.com/ischasny/ipfs-camp-routing](https://github.com/ischasny/ipfs-camp-routing).
+
+### Explore IPNI Records
+First, lets take a look at an example IPNI record.
+* Check out the Filecion Saturn website, [located at the CID bafybeia57mwbxw3csprt72a6bd6o4uedazn3vo6tv64xken6fgmaxtiugy](https://bafybeia57mwbxw3csprt72a6bd6o4uedazn3vo6tv64xken6fgmaxtiugy.ipfs.dweb.l).
```
Comment (walkerlj0) [inline content/en/curriculum/dev-tools/network-indexer/index.md:87]:
```suggestion
* Check out the Filecion Saturn website, located at the CID `bafybeia57mwbxw3csprt72a6bd6o4uedazn3vo6tv64xken6fgmaxtiugy`.
```

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 15]
```diff
@@ -0,0 +1,328 @@
...
+With an open protocol that anyone can use, as seen here in the [go-delgated-routing](https://github.com/ipfs/go-delegated-routing) library on their server.
+to enable faster discovery and routing, along with load balancing and an alternative market where routing is done, besides the DHT
+
+Laern more about IPFS Indexer, the motivations behind it, and potential use cases, see the video **[State of Content Routing| IPFS Camp Lisbon 2022](https://www.youtube.com/watch?v=DLCTEXbF1Es)**
+
+
+## Indexer Implementations
+IPNI builds its index by processing (Advertisements)[https://github.com/ipni/storetheindex/blob/main/api/v0/ingest/schema/schema.ipldsch#L40]. The Advertisement construct allows a Storage Provider to offer their CIDs to IPNI, which would make these CIDs available for fast lookups once the Advertisement has been processed. Since Filecoin and IPFS store data on separate networks, using different methods of data transfer, there are who different ways that the information about the CIDs are communicated to IPNI.
```
Comment (walkerlj0) [inline content/en/curriculum/dev-tools/network-indexer/index.md:35]:
```suggestion
IPNI builds its index by processing [Advertisements](https://github.com/ipni/storetheindex/blob/main/api/v0/ingest/schema/schema.ipldsch#L40). The Advertisement construct allows a Storage Provider to offer their CIDs to IPNI, which would make these CIDs available for fast lookups once the Advertisement has been processed. Since Filecoin and IPFS store data on separate networks, using different methods of data transfer, there are who different ways that the information about the CIDs are communicated to IPNI.
```

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 16]
```diff
@@ -0,0 +1,328 @@
...
+
+Start the index-provider:
+
+```
+$ provider daemon
+```
+
+### See It In Action
+
+Make sure `ipfs daemon` is running and open up a WebUI at [http://127.0.0.1:5001/webui](http://127.0.0.1:5001/webui) and upload a file.
```
Comment (walkerlj0) [inline content/en/curriculum/dev-tools/network-indexer/index.md:301]:
```suggestion
Make sure `ipfs daemon` is running and open up a WebUI at [http://127.0.0.1:5001/webui](http://127.0.0.1:5001/webui) and upload a file.

You can also add a file using the `ipfs add path/to/local/file` command and take note of the CID output in the terminal
```

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 17]
```diff
@@ -0,0 +1,346 @@
...
+- deep
+---
+
+## <Dev Tool> Introduction
+The [IPNI network indexer](https://github.com/ipni) is a project created and maintained by Protocol Labs that is designed to index the data on the Filecoin and IPFS networks, and work alongside the Distributed Hash Table (DHT) employed by both the IPFS and Filecoin networks to enable the fast and efficient retrieval of content-addressed data.
+
+## Why Use Indexed Content Routing?
+Protocol Labs' Network Indexer (IPNI) enables any user or developer to query both the IPFS and Filecoin public nodes to find content-addresses data using the [CID](../ipfs/content-addressing/#content-addressing).
+
+Since IPFS and Filecoin use different protocols to retrieve data (IPFS uses Bitswap, and Filecion uses Graphsync), there is a need for a solution that makes it possible for anyone who would want to retrieve and use that data to locate and understand which protocol they can use to retrieve that data and use it in other applications.
```
Comment (ischasny) [inline content/en/curriculum/dev-tools/network-indexer/index.md:21]: Maybe add "More information about IPNI can be found here": https://github.com/ipni/specs/blob/main/IPNI.md

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 18]
```diff
@@ -0,0 +1,346 @@
...
+
+Kubo 16 introduced Delegated Routing protocol that allows a node to publish its CIDs to alternative routing systems alongside the DHT and be used for subsequent content-addressed data lookups.
+
+The Indexer also makes it possible, in a way not achieved before, for people to retrieve data from the Filecoin network, as there is a lot of data stored across Filecoin nodes, but  no way to search and retrieve that data searching either by the CID of multihash of that data.
+
+Though the DHT is an amazing, distributed way to advertise and discover content, the IPFS indexer adds another layer that can give you other options for retrieving files for your applicaitons. With an open protocol that anyone can use, as seen here in the [go-delgated-routing](https://github.com/ipfs/go-delegated-routing) library on their server.
```
Comment (ischasny) [inline content/en/curriculum/dev-tools/network-indexer/index.md:27]:
```suggestion
Though the DHT is an amazing, distributed way to advertise and discover content, IPNI adds another layer that can give you other options for retrieving files for your applications. With an open protocol that anyone can use, as seen here in the [go-delgated-routing](https://github.com/ipfs/go-delegated-routing) library on their server.
```

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 19]
```diff
@@ -0,0 +1,346 @@
...
+With an open protocol that anyone can use on their server, as seen here in the [go-delgated-routing](https://github.com/ipfs/go-delegated-routing) library.
+to enable faster discovery and routing, along with load balancing and an alternative market where routing is done, besides the DHT
+
+Laern more about IPFS Indexer, the motivations behind it, and potential use cases, see the video **[State of Content Routing| IPFS Camp Lisbon 2022](https://www.youtube.com/watch?v=DLCTEXbF1Es)**
```
Comment (ischasny) [inline content/en/curriculum/dev-tools/network-indexer/index.md:31]:
```suggestion
Laern more about IPNI, the motivations behind it, and potential use cases, see the video **[State of Content Routing| IPFS Camp Lisbon 2022](https://www.youtube.com/watch?v=DLCTEXbF1Es)**
```

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 20]
```diff
@@ -0,0 +1,346 @@
...
+[Reframe](https://github.com/ipfs/go-delegated-routing) is a Request-response Protocol (RPC) used by IPFS Kubo 16+ that allows the node to publish its own CIDs as well as to look up other CIDs in an alternative routing system such as IPNI alongside the DHT. That doesn't have to be either / or choice - content can be looked up or published to both in the same time.
+
+Kubo doesn't know how to speak to IPNI by itself. Instead it relies on someone translating reframe messages into the IPNI protocol. That is done to enable alternative routing systems market without being bound to the DHT or IPNI protocols.
+
+That job is done by Index Provider process that can be run alongside a Kubo node as a bridge to IPNI.
+With versions of kubo 16+, the Refame RPC is integrated as a feature
+* Reframe adds an additional way to discover peers, content, and IPNS records
+* With Reframe, you can configure your IPFS kubo node to publish a snapshot of all of the CIDs on your node at whatever frequency you would like
+* Reframe uses HTTP transport to store this information on an indexer node
+* Reframe is an alternative to the DHT
```
Comment (ischasny) [inline content/en/curriculum/dev-tools/network-indexer/index.md:53]:
```suggestion
* Reframe makes alternative routing systems other than DHT possible
```

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 21]
```diff
@@ -0,0 +1,346 @@
...
+* Content can be published to both the IPFS DHT and Reframe with a tool called parallel
```
Comment (ischasny) [inline content/en/curriculum/dev-tools/network-indexer/index.md:54]: This block ends abruptly, probably a typo?

File: content/en/curriculum/dev-tools/network-indexer/index.md (hunk 17 above)
Comment (ischasny) [reply content/en/curriculum/dev-tools/network-indexer/index.md:21]: Would be good to mention that cid.contain is the largest IPNI deployment to date, managed by PL. However there are other deployments too. cid.contact can be explored via a [UI](https://cid.contact) or via (API)[https://cid.contact/cid/bafybeia57mwbxw3csprt72a6bd6o4uedazn3vo6tv64xken6fgmaxtiugy]. The API call above returns a `json` object that says where and how the data represented with a CID `bafybeia57mwbxw3csprt72a6bd6o4uedazn3vo6tv64xken6fgmaxtiugy` can be fetched from.

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 22]
```diff
@@ -0,0 +1,346 @@
...
+```
+$ storetheindex init
+```
+
+Start the indexer
+
+```
+$ storetheindex daemon
+```
+
```
Comment (ischasny) [inline content/en/curriculum/dev-tools/network-indexer/index.md:121]: Might be worth to clarify:

In this tutorial we are going to run an indexer locally, however it's also possible to advertise your data to cid.contact directly. The reason why we are not going to do that now - is because cid.contact needs to be able to dial back into your node to fetch a chain of advertisements. However, that might be problematic to do behind home routers and would require some additional firewall configuration.

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 23]
```diff
@@ -0,0 +1,346 @@
...
+* Reframe adds an additional way to discover peers, content, and IPNS records
+* With Reframe, you can configure your IPFS kubo node to publish a snapshot of all of the CIDs on your node at whatever frequency you would like
+* Reframe uses HTTP transport to store this information on an indexer node
+* Reframe is an alternative to the DHT
+* Content can be published to both the IPFS DHT and Reframe with a tool called parallel
+  Reframe was created from a [kubo spec](https://github.com/ipfs/specs/blob/main/reframe/REFRAME_PROTOCOL.md) (the IPFS go implementation) that allows IPFS nodes to advertise their content to other systems besides the DHT.
+
+
+## Tutorial: IPNI
+In this simple tutorial, we are going to configure and run an IPFS node to communicate data to the indexer and take a look at what that data looks like.
```
Comment (ischasny) [inline content/en/curriculum/dev-tools/network-indexer/index.md:59]: Maybe add:

We are going to setup three things:
1. Kubo IPFS node
2. `index-provider` that is going to be used as a bridge between IPNI and IPFS
3. Network indexer that the IPFS node is going to advertise its contents to.

File: content/en/curriculum/dev-tools/network-indexer/index.md (hunk 17 above)
Comment (ischasny) [reply content/en/curriculum/dev-tools/network-indexer/index.md:21]: We can also mention some stats that at the moment of writing cid.contact has 1.3 trillion CIDs indexed that is spread across >400 [providers](https://cid.contact/providers) and ingests over 10 billion new CIDs weekly.

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 24]
```diff
@@ -0,0 +1,346 @@
...
+
+You can also add a file using the `ipfs add path/to/local/file` command and take note of the CID output in the terminal
+
+Once upoloaded you should see logs start rolling in `storetheindex` and `indexprovider` command lines.
+
+#### Check the Index
+Open a new terminal window and check whether your cid got indexed by executing
+
+```
+$ storetheindex find --cid <CID>
```
Comment (walkerlj0) [inline content/en/curriculum/dev-tools/network-indexer/index.md:328]:
```suggestion
storetheindex find --cid <CID>
```

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 25]
```diff
@@ -0,0 +1,346 @@
...
+level:
+- shallow
+- deep
+---
+
+## <Dev Tool> Introduction
+The [IPNI network indexer](https://github.com/ipni) is a project created and maintained by Protocol Labs that is designed to index the data on the Filecoin and IPFS networks, and work alongside the Distributed Hash Table (DHT) employed by both the IPFS and Filecoin networks to enable the fast and efficient retrieval of content-addressed data.
+
+## Why Use Indexed Content Routing?
+Protocol Labs' Network Indexer (IPNI) enables any user or developer to query both the IPFS and Filecoin public nodes to find content-addresses data using the [CID](../ipfs/content-addressing/#content-addressing).
```
Comment (walkerlj0) [inline content/en/curriculum/dev-tools/network-indexer/index.md:19]:
```suggestion
Protocol Labs' Network Indexer (IPNI) enables any user or developer to query both the IPFS and Filecoin public nodes to find content-addresses data using the [CID](../ipfs/content-addressing/#content-addressing). The cid.contact project is the largest IPNI deployment to date, managed by PL. However there are other deployments too. cid.contact can be explored via a [UI](https://cid.contact/) or via [API](https://cid.contact/cid/bafybeia57mwbxw3csprt72a6bd6o4uedazn3vo6tv64xken6fgmaxtiugy). 
```

Comment (walkerlj0) [reply content/en/curriculum/dev-tools/network-indexer/index.md:19]: @ischasny first part of your comment added here

File: content/en/curriculum/dev-tools/network-indexer/index.md (hunk 17 above)
Comment (walkerlj0) [reply content/en/curriculum/dev-tools/network-indexer/index.md:21]: I want to avoid adding numbers & istats that are changing -- it would mean we would have to go in and update this document almost weekly, but we can mention general things like 'The largest at this time' which won't change. Is there a place these stats are published that we can point to instead?

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 26]
```diff
@@ -0,0 +1,346 @@
...
+
+Since IPFS and Filecoin use different protocols to retrieve data (IPFS uses Bitswap, and Filecion uses Graphsync), there is a need for a solution that makes it possible for anyone who would want to retrieve and use that data to locate and understand which protocol they can use to retrieve that data and use it in other applications.
+
+Kubo 16 introduced Delegated Routing protocol that allows a node to publish its CIDs to alternative routing systems alongside the DHT and be used for subsequent content-addressed data lookups.
+
+IPNI also makes it possible, in a way not achieved before, for people to retrieve data from the Filecoin network, as there is a lot of data stored across Filecoin nodes, but  no way to search and retrieve that data searching either by the CID of multihash of that data.
```
Comment (walkerlj0) [inline content/en/curriculum/dev-tools/network-indexer/index.md:25]:
```suggestion
IPNI also makes it possible, in a way not achieved before, for people to retrieve data from the Filecoin network, as there is a lot of data stored across Filecoin nodes, but  no way to search and retrieve that data searching either by the CID of multihash of that data. The API calls return json objects that contain information about where and how the data represented with a CID such as `bafybeia57mwbxw3csprt72a6bd6o4uedazn3vo6tv64xken6fgmaxtiugy` can be fetched from.
```

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 27]
```diff
@@ -0,0 +1,346 @@
...
+
+Kubo doesn't know how to speak to IPNI by itself. Instead it relies on someone translating reframe messages into the IPNI protocol. That is done to enable alternative routing systems market without being bound to the DHT or IPNI protocols.
+
+That job is done by Index Provider process that can be run alongside a Kubo node as a bridge to IPNI.
+With versions of kubo 16+, the Refame RPC is integrated as a feature
+* Reframe adds an additional way to discover peers, content, and IPNS records
+* With Reframe, you can configure your IPFS kubo node to publish a snapshot of all of the CIDs on your node at whatever frequency you would like
+* Reframe uses HTTP transport to store this information on an indexer node
+* Reframe makes alternative routing systems other than DHT possible
+* Content can be published to both the IPFS DHT and Reframe with a tool called parallel
```
Comment (walkerlj0) [inline content/en/curriculum/dev-tools/network-indexer/index.md:54]:
```suggestion
* Content can be published to both the IPFS DHT and Reframe with a tool called ParallelHelper
```

File: content/en/curriculum/dev-tools/network-indexer/index.md (hunk 21 above)
Comment (walkerlj0) [reply content/en/curriculum/dev-tools/network-indexer/index.md:54]: Changed it to say ParallelHelper

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 28]
```diff
@@ -0,0 +1,346 @@
...
+  Reframe was created from a [kubo spec](https://github.com/ipfs/specs/blob/main/reframe/REFRAME_PROTOCOL.md) (the IPFS go implementation) that allows IPFS nodes to advertise their content to other systems besides the DHT.
+
+
+## Tutorial: IPNI
+In this simple tutorial, we are going to configure and run an IPFS node to communicate data to the indexer and take a look at what that data looks like.
+
+#### Workshop: Indexed Content Routing| IPFS Camp Lisbon 2022
+See how to set up a local indexer and index-provider in this workshop from IPFS Camp 2022.
```
Comment (walkerlj0) [inline content/en/curriculum/dev-tools/network-indexer/index.md:62]:
```suggestion
See how to set up a local indexer and index-provider in this workshop from IPFS Camp 2022. In this tutorial you will: 
* Modify the config for your kubo IPFS node
* Set up a local network indexer that the IPFS node is going to advertise its contents to
* Set up index-provider that is going to be used as a bridge between IPNI and IPFS

```

File: content/en/curriculum/dev-tools/network-indexer/index.md (hunk 23 above)
Comment (walkerlj0) [reply content/en/curriculum/dev-tools/network-indexer/index.md:59]: See below

File: content/en/curriculum/dev-tools/network-indexer/index.md (hunk 22 above)
Comment (walkerlj0) [reply content/en/curriculum/dev-tools/network-indexer/index.md:121]: Added to top of this part of the conversation -- I thought that index-provider solved problems posed by home firewall configurations?

File: content/en/curriculum/dev-tools/network-indexer/index.md [hunk 29]
```diff
@@ -0,0 +1,346 @@
...
+```
+https://cid.contact/cid/<CID>
+```
+
+> This one is located at [https://cid.contact/cid/bafybeia57mwbxw3csprt72a6bd6o4uedazn3vo6tv64xken6fgmaxtiugy](https://cid.contact/cid/bafybeia57mwbxw3csprt72a6bd6o4uedazn3vo6tv64xken6fgmaxtiugy)
+
+* Take a look at the providers who are broadcasting CIDs to the network at [https://cid.contact/providers](https://cid.contact/providers)
+
+### Set up a Local Indexer
+Installing and running the indexer on your local machine makes it possible for your kubo node to communicate with the indexer, by creating the advertisements in a format the indexer can ingest.
```
Comment (walkerlj0) [inline content/en/curriculum/dev-tools/network-indexer/index.md:100]:
```suggestion
Installing and running the indexer on your local machine makes it possible for your kubo node to communicate with the indexer, by creating the advertisements in a format the indexer can ingest.

In this tutorial we are going to run an indexer locally, however it's also possible to advertise your data to cid.contact directly. Running a local node makes it possible for cid.contact to dial back into your node to fetch a chain of advertisements. This is possible without extra firewall configurations because of the provider-node setup in the next sections.
```

File: content/en/curriculum/dev-tools/network-indexer/index.md (hunk 17 above)
Comment (ischasny) [reply content/en/curriculum/dev-tools/network-indexer/index.md:21]: Not at the moment. That's ok too.

**Review by ischasny — APPROVED** [review]
